lib/sqlalchemy/sql/elements.py

```diff
@@ -2499,21 +2542,25 @@ def columns(self, *cols, **types):
         )
 
     @property
-    def type(self):
+    def type(self) -> NullType:
         return type_api.NULLTYPE
 
     @property
-    def comparator(self):
-        return self.type.comparator_factory(self)
+    def comparator(self) -> NullType.Comparator[Any]:
+        return self.type.comparator_factory(self)  # type: ignore [arg-type]
```

Comment:
Mypy: `Argument 1 has incompatible type "TextClause"; expected "ColumnElement[_T]"` Complete mystery

Reply:
https://github.com/GodelTech/sqlalchemy/blob/ced96d1f1d75a5a0171bf5942d497011f8fdb297/lib/sqlalchemy/sql/type_api.py#L233
https://github.com/GodelTech/sqlalchemy/blob/ced96d1f1d75a5a0171bf5942d497011f8fdb297/lib/sqlalchemy/sql/type_api.py#L98-L100
But `TextClause` doesn't inherit from `ColumnElement`. Maybe it's worth to loosen `expr: ColumnElement[_T]` up to `ClauseElement`?

Reply:
Will probably not change anything in `sql.type_api`, but this is helpful - will definitely leave it in a comment